async-commit-the-accelerator-for-transaction-commit-in-tidb-5.0.md

```diff
@@ -0,0 +1,326 @@
+
+The following diagram shows the commit process in TiDB before Async Commit is introduced.
+
+![The commit process before Async Commit is introduced](media/commit-process-before-ascyn-commit.png)
+
+<div class="caption-center"> The commit process before Async Commit is introduced </div>
+
+After the user sends a COMMIT statement to TiDB, TiDB has to go through at least the following steps before returning the commit result to the user:
+
+1. Concurrently prewrites all keys.
+2. Gets a timestamp from Placement Driver (PD) as the commit timestamp (Commit TS).
+3. Commits the primary key.
+
+The critical path for the entire commit process includes at least two round trips between TiDB and TiKV. Only secondary keys are committed asynchronously in the background.
+
+In TiDB's transaction model, TiDB nodes are like the coordinators of transactions, and TiKV nodes are the participants. In common two-phase commit implementations, the coordinator's data is stored locally, while in TiDB's transaction model, all transaction-related data is stored on TiKV. Therefore, theoretically the state of a transaction is determined at the end of the first phase, but TiDB needs to complete part of the second phase—storing the state of the transaction on TiKV—before it can return the determined transaction state to the user.
+
+However, this also means that there is room for improvement in TiDB's original transaction model. Can the Percolator transaction model be improved so that the state of a transaction can be determined without additional remote procedure calls (RPCs) after the first phase is completed?
+
+## Determine the transaction state earlier
+
+Before the introduction of Async Commit, a transaction commit was successful only after the primary key of the transaction was committed. The aim of Async Commit is to advance the determination of the transaction state to the moment when prewrites are completed. In this way, the entire second phase of the commit can be done asynchronously. For an Async Commit transaction, this means that the commit is successful as long as all the keys in the transaction have been successfully prewritten.
+
+The following diagram shows the commit process of an Async Commit transaction. You can see that the original part of getting the Commit TS is missing. Instead, TiDB gets a timestamp from PD as the Min Commit TS before the prewrites. The reasons for the change will be explained in the following sections.
+
+![The commit process of an Async Commit transaction](media/commit-process-of-an-async-commit-transaction.png)
+
+<div class="caption-center"> The commit process of an Async Commit transaction </div>
+
+To reach the goal of advancing the time when the transaction state is determined, there are two main issues that need to be addressed:
+
+* How to determine if all keys have been prewritten?
+* How to determine the Commit TS of a transaction?
```

Comment:
```suggestion
* How to determine the Commit TS of a transaction
```

Reply:
Updated in https://github.com/pingcap/blog/pull/581/commits/54d3197c266d03f140d0abd3c62b1c4261eda1a3.